base/sort.jl

```diff
@@ -415,9 +415,9 @@ sort(v::AbstractVector; kws...) = sort!(copymutable(v); kws...)
 
 ## other iterables and fallback ##
 
-sort(s::String; kws...) = String(sort(collect(s); kws...))
+sort(s::String; kws...) = String(sort!(collect(s); kws...))
 sort(n::Number) = throw(MethodError(sort, n))
-sort(itr; kws...) = sort(collect(itr); kws...)
+sort(itr; kws...) = sort!(collect(itr); kws...)
```

Comment:
I hope `collect` never aliases. x-ref discussion in https://github.com/JuliaLang/julia/pull/15546